homework/index.js

```diff
@@ -30,7 +30,47 @@
   }
 
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const li = createAndAppend('li', ul, {
+      class: 'container',
+    });
+
+    if (repo.name != null) {
+      createAndAppend('li', li, {
+        text: 'Repository :',
+        class: 'leftHandSide',
+      });
+      const repositoryName = createAndAppend('li', li, {
+        class: 'rightHandSide',
+      });
+      createAndAppend('a', repositoryName, {
+        href: repo.html_url,
+        target: '_blank',
+        text: repo.name,
+      });
+    }
+    function keyToRead(key) {
+      let newValue = key;
+      for (let i = 0; i < key.length; i++) {
+        if (key[i] === '_') {
+          newValue = newValue.substr(0, i) + ' ' + newValue.substr(i + 1);
+        }
+      }
+      return newValue[0].toUpperCase() + newValue.slice(1);
+    }
+
+    const renderKeys = ['description', 'forks', 'updated_at'];
+    renderKeys.forEach(key => {
+      if (repo[key] != null) {
+        createAndAppend('li', li, {
+          text: keyToRead(key) + ' :',
+          class: 'leftHandSide',
+        });
+        createAndAppend('li', li, {
+          text: repo[key],
+          class: 'rightHandSide',
+        });
+      }
+    });
```

Comment:
I'm looking at the lines 51-73. I gather you attempted to make your code DRYer by using a loop rather than repeating the same code three times. However, using a loop makes your code unnecessary complicated. And your attempt to convert the property names of the repository object into label text that you can display further complicates the code.

The standard tool for making your code DRY is to use functions. For instance, let's assume that you use an HTML `table` for rendering the repository details. Then you can write an `addRow()` function that you could call like this:

```js
addRow(table, 'Description', repo.description);
addRow(table, 'Forks', repo.forks);
addRow(table, 'Updated at', repo.updated_at);
```

Voila: no need for a loop or a function to convert property names into label text.